tsl/src/continuous_aggs/invalidation.c

```diff
@@ -659,8 +674,8 @@ invalidation_entry_try_merge(Invalidation *entry, const Invalidation *newentry)
 }
 
 static void
-cut_and_insert_new_cagg_invalidation(const CaggInvalidationState *state, const Invalidation *entry,
-									 int32 cagg_hyper_id)
+cut_and_insert_new_cagg_invalidation(const HypertableInvalidationState *state,
```

Comment:
I'm a bit confused by the name of this function and wonder if it still implies what this function is supposed to do. First, would it make more sense to have `CAggInvalidationState` as first parameter instead of `HypertableInvalidationState`, similar to `cut_cagg_invalidation`? I see that only `state->cagg_log_rel` is needed and that can work with `CAggInvalidationState` too. But I don't really know much about the places where this function is called, so `HypertableInvalidationState` may make more sense depending on the context.

Also, does it really cut an invalidation?

Reply:
The state here is the `HypertableIvalidationState` because we're moving the hypertable invalidation logs to the materialization invalidation logs. See the caller.

Reply:
The name is a bit strange since it does not really cut anything, as you say; it just inserts the entry into the materialization log. But as @fabriziomello says, this belongs to the first phase of the refresh (moving hypertable invalidations to materialization invalidations), so it should use `HypertableInvalidationState`.